experimental/host/mt7688/pin.go

```diff
@@ -27,67 +26,58 @@ const (
-func initPins() {
-	for name := range mappingMT7688 {
-		cpuPin := cpuPins[name]
-
-		// Initialize sysfs access right away.
-		cpuPin.sysfsPin = sysfs.Pins[cpuPin.number]
-	}
+var cpuPins = []Pin{
+	// TODO: discover default pull
+	{number: 0, name: "GPIO0", defaultPull: gpio.Float},
+	{number: 1, name: "GPIO1", defaultPull: gpio.Float},
+	{number: 2, name: "GPIO2", defaultPull: gpio.Float},
+	{number: 3, name: "GPIO3", defaultPull: gpio.Float},
+	{number: 4, name: "GPIO4", defaultPull: gpio.Float},
+	{number: 5, name: "GPIO5", defaultPull: gpio.Float},
+	{number: 6, name: "GPIO6", defaultPull: gpio.Float},
+	{number: 7, name: "GPIO7", defaultPull: gpio.Float},
+	{number: 8, name: "GPIO8", defaultPull: gpio.Float},
+	{number: 9, name: "GPIO9", defaultPull: gpio.Float},
+	{number: 10, name: "GPIO10", defaultPull: gpio.Float},
+	{number: 11, name: "GPIO11", defaultPull: gpio.Float},
+	{number: 12, name: "GPIO12", defaultPull: gpio.Float},
+	{number: 13, name: "GPIO13", defaultPull: gpio.Float},
+	{number: 14, name: "GPIO14", defaultPull: gpio.Float},
+	{number: 15, name: "GPIO15", defaultPull: gpio.Float},
+	{number: 16, name: "GPIO16", defaultPull: gpio.Float},
+	{number: 17, name: "GPIO17", defaultPull: gpio.Float},
+	{number: 18, name: "GPIO18", defaultPull: gpio.Float},
+	{number: 19, name: "GPIO19", defaultPull: gpio.Float},
+	{number: 20, name: "GPIO20", defaultPull: gpio.Float},
+	{number: 21, name: "GPIO21", defaultPull: gpio.Float},
+	{number: 22, name: "GPIO22", defaultPull: gpio.Float},
+	{number: 23, name: "GPIO23", defaultPull: gpio.Float},
+	{number: 24, name: "GPIO24", defaultPull: gpio.Float},
+	{number: 25, name: "GPIO25", defaultPull: gpio.Float},
+	{number: 26, name: "GPIO26", defaultPull: gpio.Float},
+	{number: 27, name: "GPIO27", defaultPull: gpio.Float},
+	{number: 28, name: "GPIO28", defaultPull: gpio.Float},
+	{number: 29, name: "GPIO29", defaultPull: gpio.Float},
+	{number: 30, name: "GPIO30", defaultPull: gpio.Float},
+	{number: 31, name: "GPIO31", defaultPull: gpio.Float},
+	{number: 32, name: "GPIO32", defaultPull: gpio.Float},
+	{number: 33, name: "GPIO33", defaultPull: gpio.Float},
+	{number: 34, name: "GPIO34", defaultPull: gpio.Float},
+	{number: 35, name: "GPIO35", defaultPull: gpio.Float},
+	{number: 36, name: "GPIO36", defaultPull: gpio.Float},
+	{number: 37, name: "GPIO37", defaultPull: gpio.Float},
+	{number: 38, name: "GPIO38", defaultPull: gpio.Float},
+	{number: 39, name: "GPIO39", defaultPull: gpio.Float},
+	{number: 40, name: "GPIO40", defaultPull: gpio.Float},
+	{number: 41, name: "GPIO41", defaultPull: gpio.Float},
+	{number: 42, name: "GPIO42", defaultPull: gpio.Float},
+	{number: 43, name: "GPIO43", defaultPull: gpio.Float},
+	{number: 44, name: "GPIO44", defaultPull: gpio.Float},
+	{number: 45, name: "GPIO45", defaultPull: gpio.Float},
+	{number: 46, name: "GPIO46", defaultPull: gpio.Float},
 }
 
+// Pin is a GPIO number (GPIOnn) on MT7688(AN/KN).
 type Pin struct {
 	// Immutable.
 	number      int
```

Comment:
[optional style comment]

An option is to inline alt as:
```
  alt [3]pin.Func
```

so that you would inline the functions inside the cpuPins definition directly. In practice it doesn't change much things, so it's just FYI as an idea.

Reply:
I had a look into this and I decided to go with the common conventions - more for new members perusing across multiple host packages.
A mapping of some sort seems to appear in most host packages.

Again, if you would like to converge convention on storing the alt functions in the Pin struct for encapsulation, happy to work that into another PR :smile: